### PR TITLE
NuGet UX Work for Multiple Versions of NuGet

### PR DESCRIPTION
**Pull request by kkdawkins**

### CODE REVIEW

File: Tasks/Common/nuget-task-common/module.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     "messages": {
         "NGCommon_IgnoringNuGetExtensionsPath": "Detected NuGet extensions loader path (NUGET_EXTENSIONS_PATH environment variable). Extensions are ignored when using the built-in NuGet client",
         "NGCommon_DetectedNuGetExtensionsPath": "Detected NuGet extensions loader path (NUGET_EXTENSIONS_PATH environment variable): %s",
-        "NGCommon_UnableToFindTool": "Unable to find tool %s"
+        "NGCommon_UnableToFindTool": "Unable to find tool %s",
+        "NGCommon_UnabletoDetectNuGetVersion" : "Unknown NuGet version selected."
```
Comment (jmyersmsft) [inline Tasks/Common/nuget-task-common/module.json]: You'll need to duplicate this message into the individual task.json files as well, unfortunately

**Comment by jmyersmsft** [comment]
:shipit:

**Comment by jmyersmsft** [comment]
Actually, I thought of one more thing, sorry. We should also add the credential provider to the tasks at the same time

File: Tasks/NugetPublisher/task.json [hunk 2]
```diff
@@ -121,6 +127,7 @@
         "NotARegularFile": "%s is not a regular file. Check the 'Path/Pattern to nupkg' property of the NuGetPublisher task.",
         "UnknownFeedType": "Unknown feed type '%s'",
         "CouldNotFindNuGetService": "Could not find the NuGet service. This task will be unable to authenticate to feeds hosted in your Team Services account/TFS project collection.",
+        "NoNuGetSpecified" : "If 'external' is selected, you must provide a path to NuGet.exe",
```
Comment (zjrunner) [inline Tasks/NugetPublisher/task.json]: field is now called 'custom'.  rename from 'external'

Comment (kkdawkins) [inline Tasks/NugetPublisher/task.json]: Done. Thanks for your feedback!

**Pull request merged**